graphistry/umap_utils.py

```diff
@@ -229,10 +280,8 @@ def _process_umap(
         """
             Returns res mutated with new _xy
         """
-        _, _, umap = lazy_umap_import_has_dependancy()
-        # need this function to use memoize
-        res._umap = umap.UMAP(**umap_kwargs)
-
+        res._umap = self._umap
```

Comment:
@silkspace  is reusing `self._umap` here right?

Reply:
afaict `def umap(...)` calls lazy_init, which calls `self.umap = UMAP(**umap_kwargs)` , so indeed not needed here